raiden/tests/unit/channel_state/test_coop_settle.py

```diff
@@ -0,0 +1,448 @@
+        pseudo_random_generator=pseudo_random_generator,
+        block_number=block_number,
+    )
+
+    assert (
+        search_for_item(
+            iteration.events,
+            SendWithdrawConfirmation,
+            {"total_withdraw": 100, "nonce": 2, "expiration": expiration},
+        )
+        is not None
+    )
+    assert (
+        search_for_item(
+            iteration.events,
+            SendWithdrawRequest,
+            {"total_withdraw": 70, "nonce": 1, "coop_settle": False, "expiration": expiration},
+        )
+        is not None
+    )
+
+    # our view
+    _assert_coop_settle_state(
+        channel_state.partner_state,
+        channel_state.our_state,
+        our_initial_model=partner_model1,
+        our_current_nonce=1,
+        coop_settle_expiration=expiration,
+        initiated_coop_settle=True,
+    )
+    # partner's view
+    _assert_coop_settle_state(
+        channel_state.our_state,
+        channel_state.partner_state,
+        our_initial_model=our_model1,
+        our_current_nonce=2,
+        coop_settle_expiration=expiration,
+        initiated_coop_settle=False,
+    )
+
+
+def test_contract_event():
+    pseudo_random_generator = random.Random()
+
+    our_model1, _ = create_model(balance=70)
+    partner_model1, privkey2 = create_model(balance=100)
+    partner_signer = LocalSigner(privkey2)
+    channel_state = create_channel_from_models(our_model1, partner_model1, privkey2)
+    block_hash = make_block_hash()
+    block_number = 1
+    expiration = 1000
+
+    channel_state.our_state.withdraws_pending[our_model1.balance] = PendingWithdrawState(
+        total_withdraw=our_model1.balance, expiration=expiration, nonce=0
+    )
+    channel_state.our_state.initiated_coop_settle = CoopSettleState(
+        total_withdraw_participant=our_model1.balance,
+        total_withdraw_partner=partner_model1.balance,
+        expiration=expiration,
+    )
+
+    withdraw_request = _make_receive_coop_settle_withdraw_request(
+        channel_state,
+        pseudo_random_generator,
+        partner_signer,
+        nonce=1,
+        total_withdraw=partner_model1.balance,
+        withdraw_expiration=expiration,
+    )
+
+    iteration = channel.state_transition(
+        channel_state=channel_state,
+        state_change=withdraw_request,
+        block_hash=block_hash,
+        pseudo_random_generator=pseudo_random_generator,
+        block_number=block_number,
+    )
+
+    assert (
+        search_for_item(
+            iteration.events,
+            SendWithdrawConfirmation,
+            {"total_withdraw": 100, "nonce": 1, "expiration": expiration},
+        )
+        is not None
+    )
+
+    assert len(iteration.events) == 1
+
+    # our view
+    _assert_coop_settle_state(
+        channel_state.our_state,
+        channel_state.partner_state,
+        our_initial_model=our_model1,
+        our_current_nonce=1,
+        coop_settle_expiration=expiration,
+        initiated_coop_settle=True,
```

Comment:
This is not clear to me. We're setting `initiated_coop_settle` to `True`, even though it was the partner who sent the coop settle withdraw request. Could this be because we set the `CoopSettleState` manually? I'm confused.